src/openassetio-core/include/openassetio/BatchElementError.hpp

```diff
@@ -116,6 +116,24 @@ class BatchElementError final {
   Str message;
 };
 
+/**
+ * Compares two batch element error instances to see if they match.
+ *
+ * @return `true` if the batch elements errors match, otherwise `false`.
+ */
+constexpr bool operator==(const BatchElementError& lhs, const BatchElementError& rhs) {
```

Comment:
[question] Curious why you chose a free function rather than a member function?

Reply:
Scott Meyers one of the known and trusted C++ members seems to think that non-member functions increase encapsulation: https://www.drdobbs.com/cpp/how-non-member-functions-improve-encapsu/184401197

Whilst investigating, I have also found these stack overflow references:

https://stackoverflow.com/a/24938533/2682142
https://stackoverflow.com/a/4622467/2682142

Reply:
Fwiw, We had made a note when estimating the issue to do it as a member function for consistency with the other exiting implementations. Also, in other classes it has to consider private members.

Reply:
The Meyers article has a compelling argument, which boils down to: when deciding the impact of changing the private API you have to consider all member functions. So removing member functions, where possible, reduces work when redesigning/refactoring the private API. I.e. member functions that only access the public API just reduces encapsulation.

Reply:
Totally makes sense, just wondering for this specific case, does the complexity of friending undo that - and do we want to have diverging approaches across the code base?

Reply:
The downside as I see it is discoverability/readability.  I think for this super-simple class with no real moving parts, a member function is probably slightly better for that reason.  But I think the encapsulation argument is definitely worth keeping in mind in future designs.

Reply:
@lpapp-foundry Sorry to be a total pain, would you mind making this a member function so it is consistent with the other classes we have so far? Thanks!

Reply:
Done, thanks.